tests/MongoDB.Driver.Tests/UpdateDefinitionBuilderTests.cs

```diff
@@ -238,6 +238,18 @@ public void Inc_Typed()
             Assert(subject.Inc("Age", 1), "{$inc: {age: 1}}");
         }
 
+        [Fact]
+        public void Indexed_Positional_Typed()
```

Comment:
below I only added one new test: `Indexed_Typed_Incorrectly`. All other changes, it's just reordering of tests to be sorted alphabetically

Reply:
>one new test:

+ AssertThrow methods

Reply:
I'm often torn about reordering existing code alphabetically.

It makes diffs hard to do.